### ISSUE TITLE
loss goes to nan

### ISSUE THREAD

**Issue by elenina5** [jaredleekatzman/DeepSurv#14]
I am working on the public datasets a this link: https://vincentarelbundock.github.io/Rdatasets/datasets.html
in particular I have tried "aids2" and "flchain". 
Nevertheless in both cases log takes the following values: 
{'best_params': None,
 'best_params_idx': -1,
 'best_valid_ci': 0.5,
 'best_validation_loss': inf,
 'train': [array(28340558372.044426),
  array(nan),
  array(nan),
  array(nan),
  array(nan)],
 'train_ci': [0.5, 0.5, 0.5, 0.5, 0.5],
 'valid': [array(nan), array(nan), array(nan), array(nan), array(nan)],
 'valid_ci': [0.5, 0.5, 0.5, 0.5, 0.5]}
what can be done to solve this issue?

**Comment by pvaezi** [comment]
I have similar issue with a different dataset! Any comments on this is greatly appreciated!

**Comment by dareneiri** [comment]
@elenina5 @pvaezi I recommend trying to normalize your features. Say for example you are working with petal width, length, and height. Normalize these to have a value ranging from 0-1. I use Sklearn's method: 
http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html

**Comment by pvaezi** [comment]
@dareneiri thanks for the comment, that worked for me! I was using StandardScaler() before. Any specific reason why standard scaler wouldn't work properly here?

**Comment by dareneiri** [comment]
@pvaezi I think this will help illustrate the differences: 
http://sebastianraschka.com/Articles/2014_about_feature_scaling.html#standardization-and-min-max-scaling

StandardScaler() will result in the distribution of the data to have a mean of 0 and SD of 1. But the values will still not be constrained to 0-1. MinMax() allows you to specify the values to only be between 0-1, which is what I think you had in mind.

**Comment by pvaezi** [comment]
@dareneiri thanks for the quick response and the link, I understand the differences between them, I was merely pondering why StandardScalar works properly on linear Cox model from lifeline library, but not here!

My main concern about MinMaxScalar is that it is sensitive to the outliers!

**Comment by dareneiri** [comment]
@pvaezi I see. The Lifeline library can use either input of features between 0-1, and features outside that range. DeepSurv uses a neural network combined with the cox model and generally with neural networks you want to pass values that are between 0-1 as the input.

**Comment by pvaezi** [comment]
@dareneiri I see! Thanks for the explanation and useful discussion!
